app/models/pay/stripe/subscription.rb

```diff
@@ -301,6 +301,20 @@ def swap(plan, **options)
         raise Pay::Stripe::Error, e
       end
 
+      # Creates a meter event to bill for usage
+      #
+      # create_meter_event(:api_request, value: 1)
+      # create_meter_event(:api_request, token: 7)
+      def create_meter_event(event_name, **payload)
+        ::Stripe::Billing::MeterEvent.create({
+          event_name: event_name,
+          payload: {
+            stripe_customer_id: customer.processor_id,
+            **payload
```

Comment:
This would hit the API twice (one to load the Stripe::Customer, one to create the MeterEvent), so we should only do that if the processor_id does not exist since we lazy create customers.

```suggestion
        api_record unless processor_id?
        ::Stripe::Billing::MeterEvent.create({
          event_name: event_name,
          payload: {
            stripe_customer_id: processor_id,
            **payload
```

Reply:
Ahh thanks, I missed that `processor_id` was available on the subscription model itself. Updated now.